expertise/execute_pipeline.py

```
@@ -28,64 +28,77 @@
+        }
+        bucket_name = destination_prefix.split('/')[2]
+        blob_prefix = '/'.join(destination_prefix.split('/')[3:])
+        gcs_client = storage.Client()
+        bucket = gcs_client.bucket(bucket_name)
+
+        print('Loading model artifacts')
+        load_model_artifacts()
+
+        print('Logging into OpenReview')
+        client_v1 = openreview.Client(baseurl=baseurl_v1, token=token)
+        client_v2 = openreview.api.OpenReviewClient(baseurl_v2, token=token)
+
+        print('Creating job ID')
+        job_id = shortuuid.ShortUUID().random(length=5)
+        if working_dir is None:
+            working_dir = f"/app/{job_id}"
+        os.makedirs(working_dir, exist_ok=True)
+
+        print('Creating job config')
+        validated_request = APIRequest(raw_request)
+        config = JobConfig.from_request(
+            api_request = validated_request,
+            starting_config = DEFAULT_CONFIG,
+            openreview_client= client_v1,
+            openreview_client_v2= client_v2,
+            server_config = server_config,
+            working_dir = working_dir
+        )
+
+        if working_dir is not None:
+            path_fields = ['work_dir', 'scores_path', 'publications_path', 'submissions_path']
+            config.job_dir = working_dir
+            config.dataset['directory'] = working_dir
+            for field in path_fields:
+                config.model_params[field] = working_dir
+
+        # Create Dataset and Execute Expertise
+        print('Creating dataset and executing expertise')
+        execute_create_dataset(client_v1, client_v2, config.to_json())
+        execute_expertise(config.to_json())
+    except Exception as e:
+        # Write error to single JSONL line in GCS
+        error_message = {
+            'error': str(e)
+        }
+        destination_blob = f"{blob_prefix}/error.json"
+        blob = bucket.blob(destination_blob)
+        blob.upload_from_string(json.dumps(error_message))
```

Wouldn't this make the job in the queue silently fail?

---

I raised the exception again, the exception will make the pipeline return failure in Vertex, and the queue reads the error status in Vertex and makes the queue fail